### DISCUSSION TITLE
uuid is referenced in the read me but never set

### DISCUSSION THREAD

**Discussion by rabol** [coderflexx/laravel-ticket#34] [Q&A]
Hi

I was wondering when th uuid on the ticket get initialised ?

In the readme there is this:

```
    return redirect(route('tickets.show', $ticket->uuid))
            ->with('success', __('Your Ticket Was created successfully.'));
```

**Comment by ousid** [comment]
Hi, @rabol 

The `uuid` is `null` by default, and in the example is letting you know that there's a `uuid` column you can use.

If you want to use the `uuid`, do the following:

- Create `app/Ticket.php` model in your app
- extend `Coderflex\LaravelTicket\Models\Ticket` model
- use the __Model observer__ to create to generate the `uuid` when creating a new ticket.
- use the `uuid` in the `getRouteKeyName` or in the `routes/web.php` to set the primary key to `uuid`

Here's an example:

```php
class Ticket extends \Coderflex\LaravelTicket\Models\Ticket
{
  ....

  protected function booted()
  {
     static::created(function (Model $model) {
      $model->uuid = Str::uuid();
    });
  }

  // to make the uuid primary by default
  public function getRouteKeyName()
  {
    return 'uuid';
  }
}
```

Or you can use the `routes/web.php`
```php
Route::get('/tickets/{ticket:uuid}', ...);
```
